extendr-api/src/robj.rs

```diff
@@ -1904,6 +1783,20 @@ impl ToVectorValue for &f64 {
     }
 }
 
+impl ToVectorValue for Option<f64> {
+    fn sexptype() -> SEXPTYPE {
+        REALSXP
+    }
+
+    fn to_numeric(&self) -> f64 {
+        if self.is_some() {
+            self.unwrap()
+        } else {
+            unsafe { R_NaReal }
```

Comment:
Here and throughout: I'm not sure this conversion is idiomatic. There is no valid Rust `f64` representation of `NA`. Sure we can return `R_NaReal`, but the calculations with it will be wrong. (This is even worse for integers.) I think the right thing to do is to panic, which then gets caught and converted into an R error. Sure this may feel inconvenient, but it means people have to either properly handle options throughout or accept that their code can't handle `NA`s. That's the only way to ensure results are always correct, in my opinion.

Reply:
You are right that calling a rust function with a NA should cause an error.

(rust)
```rust
#[extendr]
my_func(x: f64) {
}
```
with (R)
```R
my_func(NA)
```
Should cause an error.

But this particular code converts an Option<> into a NA if the option is None
and is intended for a return type where you may want to signal one or more NAs.

(rust)
```rust
#[extendr]
one_or_na(x: bool) -> Option<f64> {
  if x {
     Some(1.)
  } else {
     None
  }
}
```
with (R)
```R
my_func(TRUE) -> 1
my_func(FALSE) -> NA
```

It should also work for vectors and iterators containing NAs signaled by options.

(rust)
```rust
#[extendr]
ones_or_nas(x: &[bool]) -> Robj {
  x.into_iter()
    .map(|x| if x { Some(1.) } else { None })
    .collect_robj()
}
```
with (R)
```R
my_func(c(TRUE, FALSE, TRUE)) -> [1, NA, 1]
```

Reply:
Got it, thanks!